### DISCUSSION TITLE
After Command, the button is blocked

### DISCUSSION THREAD

**Discussion by StounhandJ** [AvaloniaUI/Avalonia#7066] [Q&A] [answered]
After the first execution of the command, the button is blocked, even if tm.Items.Count is greater than zero or there is no such condition
```c#
public class TimeManageViewModel : ReactiveObject
    {
        public TimeManageViewModel()
        {
            SaveCommand = ReactiveCommand.CreateFromTask(async () =>
                {
                    foreach (var item in Items)
                    {
                        this.Controller.timeData.updateDayDate(item,
                            new DayData(
                                item,
                                DefinitionMode.None,
                                SelectedTimeStart,
                                SelectedTimeEnd,
                                DimmingValue,
                                0
                            ));
                    }

                    await this.Controller.sendTimeData();
                },
                this.WhenAnyValue(tm => tm.Items.Count, i => i != 0));
        }

        public ICommand SaveCommand { get; }
```

**Comment by workgroupengineering** [comment]
tm.Items must implement INotifyPropertyChanged  see https://www.reactiveui.net/docs/handbook/when-any/

**Reply by StounhandJ** [reply]
it implements INotifyPropertyChanged
```c#
public Collection<DateTime> Items
        {
            get => _items;
            set => this.RaiseAndSetIfChanged(ref _items, value);
```

**Reply by StounhandJ** [reply]
the problem is not with WhenAnyValue. If you remove it, the button will be available, but after a single click it is blocked and it cannot be used

**Reply by MarkusKgit** [reply]
Is it possible that your Task throws and thus kills the Observable? 
[https://www.reactiveui.net/docs/handbook/commands/#handling-exceptions](https://www.reactiveui.net/docs/handbook/commands/#handling-exceptions)

**Reply by workgroupengineering** [reply]
Collection<T> does not implement INotifyPropertyChanged, replace it with AvaloniaList<T> and add exeception handler like suggest @MarkusKgit .

**Comment by maxkatz6** [comment, marked as answer]
1. Collection must implement INotifyCollectionChanged and INotifyPropertyChanged. Examples: ObservableCollection or AvaloniaList. But if you have the same problem without WhenAny, it's not the case. 
2.  If command crashed, it won't be enabled anymore. You need to handle it properly. As it was mentioned. 
3. The most important, command is disabled while it is running. So double check that you await call actually ends.

**Comment by maxkatz6** [comment]
In overall, it's more question about ReactiveUI and not Avalonia. You can learn more here https://www.reactiveui.net/docs/handbook/commands/

**Comment by StounhandJ** [comment]
The reason for this behavior is the running of the team. The command is asynchronous, but it runs for more than 50 seconds and the button is blocked until the end of the work.

@maxkatz6 Was the closest to the right solution
